src/Sylius/Bundle/OrderBundle/Controller/OrderItemController.php

```diff
@@ -42,12 +54,184 @@ public function createNew()
     }
 
     /**
-     * Get order repository.
+     * @param Request $request
      *
+     * @return Response
+     */
+    public function addAction(Request $request)
+    {
+        $configuration = $this->requestConfigurationFactory->create($this->metadata, $request);
+
+        $this->isGrantedOr403($configuration, ResourceActions::CREATE);
+        $newResource = $this->newResourceFactory->create($configuration, $this->factory);
+
+        $this->getItemQuantityModifier()->modify($newResource, 1);
+
+        $form = $this->resourceFormFactory->create($configuration, $newResource);
+
+        if ($request->isMethod('POST') && $form->submit($request)->isValid()) {
+            $newResource = $form->getData();
+
+            $event = $this->eventDispatcher->dispatchPreEvent('sylius.cart_item.pre_create', $configuration, $newResource);
```

Comment:
There is no `CartItem` entity, so sticking to the Resource logic we should dispatch `OrderItem` pre_create event. Someone can have a listener to this event and it will not get triggered on add to cart action what is wrong.